word_counter.rb

```diff
@@ -0,0 +1,27 @@
+class WordCounter
+  def initialize
+  end
+
+  def count(text)
+    parsed = text.downcase.gsub(/(?:[^\w\s\n'\-]|(?<=[^a-z])\-|\-(?=[^a-z]))+/, '').gsub(/[\n\s]+/, ' ').strip
+    words = parsed.split(' ')
+
+    list = {}
+
+    words.each do |word|
+      if list.key?(word)
+        list[word] += 1
+      else
+        list[word] = 1
+      end
```

Comment:
Ah. I'm guessing ruby supports assignment if a key doesn't exist? E.g.

```ruby
words.each do |word|
  list[word] += 1
end
```

Also, I'm not sure what you mean about looking into other ways of using `.each`. Any suggestions on what to look for?